spec/models/concerns/updateable_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+require "#{Rails.root}/app/models/concerns/updateable"
+
+describe Updateable do
+  describe '#update_params' do
+    it 'correctly generates an array of parameter rules' do
+      permitted_params = Resource.update_params
+
+      expect(permitted_params).to include(:time)
+      expect(permitted_params).to include(tips: [])
+    end
+
+    it 'correctly generates for has and belongs to many relationships' do
+      permitted_params = QuizResponse.update_params
```

Comment:
This creates a dependency on existing models in this project, which I don't like, but I don't think it's a pressing issue. I'll think through how to remove that dependency at some point.

Reply:
If you wanted to make this generalized within the test, you could declare a new fake active record object and use that:

```
class NonExistentModel < ApplicationRecord
  has_and_belongs_to_many :questionable_assertions
end

expect(NonExistentModel.update_params).to include(questionable_assertions: [])
```

I think you'd have to solve for the actual Updateable module in that case, which I think you'd probably want `.reflect_on_all_associations` for.

_All that said_, if the permissions are wide open anyway, it may be easier just to use `params.permit!` in any case.

Reply:
Sorry could you clarify what I would use `.reflect_on_all_associations` for?

Reply:
Also I just remembered that I tried to do what you suggested above, but I get an error because the model in the test doesn't actually exist in the database schema. Do you know if there's something I can do to get that to work?

Reply:
Ah, so you'd need to make sure it doesn't go back to the database, probably when you do `self.attributes`. Maybe something like this?

```
class NonExistentModel < ApplicationRecord
  has_and_belongs_to_many :questionable_assertions

  def self.attributes
    [:one, :two, :three]
  end
end
```

The intent of the fake class is to be able to let Updateable automatically include HABTM relations, so you don't need the `params << { mindset_ids: [] }` as above. Then using the fake class here would let you test Updateable independently of any of the real models in the app, and make the test code more explicit:

```
expect(update_params).to include(:one, :two, :three)
expect(update_params[:questionable_assertions]).to eq([])
```